client/src/components/AddItemFormSteps/StepThree.jsx

```diff
@@ -0,0 +1,197 @@
+import { Link } from "react-router-dom";
+import { useEffect, useState } from "react";
+import {
+  myAccountPath,
+  profilePath,
+  shopProductPath,
+} from "../../utilities/paths";
+import Select from "react-select";
+import { customStyles } from "../../utilities/selectStyle";
+import { countryList } from "../../utilities/countryList";
+import { postProduct } from "../../utilities/productsApi";
+import { useNavigate } from "react-router-dom";
+import moment from "moment";
+import { DATETIME_FORMAT } from "../../utilities/constants";
+import { postImagesToCloudinary } from "../../utilities/cloudinaryApi";
+import LoadingSpinner from "../LoadingSpinner";
+import { ToastContainer, toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import { useUserContext } from "../../contexts/UserContextProvider";
+
+const StepThree = ({
+  prevStep,
+  productDetails,
+  setProductDetails,
+  handleInputData,
+  images,
+}) => {
+  const { user } = useUserContext();
+  const [isLoading, setIsLoading] = useState(false);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    setProductDetails({
+      ...productDetails,
+      address: user.address === null ? "" : user.address,
+      email: user.email,
+      phone: user.phone === null ? "" : user.phone,
+      city: user.city === null ? "" : user.city,
+      country: user.country === null ? "" : user.country,
+      zipCode: user.zipCode === null ? "" : user.zipCode,
+    });
+  }, []);
+
+  const handleCountryChange = (selectedOption) => {
+    setProductDetails({ ...productDetails, country: selectedOption.value });
+  };
+
+  const handlePostItem = async () => {
+    setIsLoading(true);
+    let imageUrls = [];
+    try {
+      for (let i = 0; i < images.length; i++) {
+        const imageData = new FormData();
+        imageData.append("file", images[i]);
+        imageData.append(
+          "upload_preset",
+          process.env.REACT_APP_CLOUDINARY_PRESET_NAME
+        );
+
+        const response = await postImagesToCloudinary(imageData);
+
+        imageUrls.push(response.data.url);
+      }
+
+      const formData = productDetails;
+      formData.images = imageUrls.join();
+      formData.startDate = moment(productDetails.startDate).format(
+        DATETIME_FORMAT
+      );
+      formData.endDate = moment(productDetails.endDate).format(DATETIME_FORMAT);
+      formData.userId = user.id;
+
+      const res = await postProduct(formData);
+      navigate(shopProductPath + `/${res.data.id}`);
+    } catch (error) {
+      setIsLoading(false);
+      toast.error("Something went wrong!", {
```

Comment:
Something went wrong is not really great message to show to user. 
Perhaps, Adding new product failed, please try again. or smtg like that